<launch> 
					<!-- Gazebo-->
    <include file="$(find rrt_learn_gazebo)/launch/teresa_gazebo.launch"/>
  <node pkg="tf" type="static_transform_publisher" name="odom_to_rgbd_cam" args="0.0 -2 0.677 0 0 0 odom camera_link 100"/>
  <node pkg="tf" type="static_transform_publisher" name="base_link_to_left_wheel" args="0.0 -0.252 0.0725 0 0 0 base_link left_wheel_link 100"/>
  <node pkg="tf" type="static_transform_publisher" name="base_link_to_right_wheel" args="0.0 0.252 0.0725 0 0 0 base_link right_wheel_link 100"/>

<node name="map_server" pkg="map_server" type="map_server" args="$(find rrt_learn_gazebo)/cfg/map2.yaml">
        <param name="frame_id" value="/map" />
    </node>

  <node pkg="laser_assembler" type="laser_assembler" name="laser_assembler" args="/scanback /scanfront">
  </node>

<!--include file="$(find active_perception_controller)/launch/laser_assemble.launch" /-->
					<!-- AMCL-->
    <include file="$(find rrt_learn_gazebo)/launch/amcl.launch" />

					<!-- RVIZ-->
<node pkg="rviz" name="rviz" type="rviz" args="-d $(find rrt_learn_gazebo)/cfg/simulation.rviz"/>

					<!-- MOVE BASE-->
    <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
        <rosparam file="$(find rrt_learn_gazebo)/cfg/costmap_common_params.yaml" command="load" ns="global_costmap" />
        <rosparam file="$(find rrt_learn_gazebo)/cfg/costmap_common_params.yaml" command="load" ns="local_costmap" />
        <rosparam file="$(find rrt_learn_gazebo)/cfg/local_costmap_params.yaml" command="load" />
        <rosparam file="$(find rrt_learn_gazebo)/cfg/global_costmap_params.yaml" command="load" />
        <rosparam file="$(find rrt_learn_gazebo)/cfg/base_local_planner_params_teb.yaml" command="load" />
        <param name="base_global_planner" type="string" value="active_perception_interface/ActivePerceptionInterface" />
    <param name="base_local_planner" value="teb_local_planner/TebLocalPlannerROS" />
    <param name="controller_frequency" value="10.0" />
        <rosparam param="recovery_behaviors">[{name: conservative_reset, type: clear_costmap_recovery/ClearCostmapRecovery},
                                              {name: aggressive_reset, type: clear_costmap_recovery/ClearCostmapRecovery}]</rosparam>
        <param name="planner_patience" type="double" value="30.0"/>
        <param name="controller_patience" type="double" value="60.0"/>
        <remap from="map" to="/map"/>
    </node>


				<!--Configuration Changer -->
    <node pkg="active_perception_controller" type="robot_config_change.py" name="config_changer" output="screen">
    </node>  

    					     <!-- RRT -->
    <node pkg="active_perception_controller" type="evaluation.py" name="evaluation" output="screen">	
	    <param name="rrt_eta" type="double" value="1.1" />
	    <param name="planning_time" type="double" value="12"/>
	    <param name="max_planning_time" type="double" value="20"/>
	    <param name="experiment_name" type="string" value="15_6_more_people"/>
	    <param name="session_name" type="string" value="full_run_new_data"/>
    </node>  		

					<!--People to costmap -->
    <node pkg="active_perception_controller" type="people_to_costmap.py" name="people_costmap" output="screen">
    </node>  

  				

</launch>
